Source/Initialization/WarpXInitData.cpp

```diff
@@ -529,3 +531,36 @@ WarpX::InitializeExternalFieldsOnGridUsingParser (
         );
     }
 }
+
+void
+WarpX::PerformanceHints ()
+{
+    // Check requested MPI ranks and available boxes
+    amrex::Long total_nboxes = 0; // on all MPI ranks
+    for (int ilev = 0; ilev <= finestLevel(); ++ilev) {
+        total_nboxes += boxArray(ilev).size();
+    }
+    if (ParallelDescriptor::NProcs() > total_nboxes)
+        amrex::Print() << "\n[Warning] [Performance] Too many resources / too little work!\n"
```

Comment:
Could this be printed in color? You can use the escape sequences that set the color.

Reply:
Hehe, good question.
In practice, escape sequences are terminal dependent and we might want to add options for this (enable, disable), because not all users know to open files with escape sequences with `less -R` for instance.

Some terminal settings also highlight keywords like error or warning by default.

Then there is the question of light vs. dark background and good color schemes.

One could use a library like [fmt](https://github.com/fmtlib/fmt) for reasonable color support. Worth considering because colors are great fun, but I would not do it in this PR yet.

Reply:
Trying to think of a way to make the message more noticeable since it will quickly get lost in the output.

Reply:
@dpgrote I agree: this is a recurring discussion that we have (e.g. this was also discussed for unused parameters). 

Alternatives that were mentioned before include aborting immediately (a bit exaggerated in this particular case), printing the message at each timestep (maybe an issue if we have many such warnings...) and including the message in a separate file (but the user may not necessarily see this).
Any other suggestion is definitely welcome, since the above ones are not ideal.

That being said, it is probably a larger issue than that particular PR, so I think it is fine to merge this one in its current state.